### PR TITLE
Marked LimitsConf and NprocConf as deprecated, use limits_conf instead

### PR DESCRIPTION
**Pull request by PaulWay**
This parser is already removed in the master branch, so it is marked as deprecated in the 1.x branch.

### CODE REVIEW

**Comment by bfahr** [comment]
@PaulWay I recommend leaving tests in place until the deprecated functionality is removed.

File: insights/parsers/nproc.py [hunk 1]
```diff
@@ -46,12 +47,20 @@ def _parse_content(content):
 @parser("limits.conf")
 class LimitsConf(Parser):
     """Parse ``limits.conf`` file contents."""
+    def __init__(cls, *args, **kwargs):
```
Comment (bfahr) [inline insights/parsers/nproc.py]: I would expect to see `self` here instead of `cls`.

File: insights/parsers/nproc.py [hunk 2]
```diff
@@ -46,12 +47,20 @@ def _parse_content(content):
...
+        super(LimitsConf, cls).__init__(*args, **kwargs)
+
     def parse_content(self, content):
         self.data = _parse_content(content)
 
 
 @parser("nproc.conf")
 class NprocConf(Parser):
     """Parse ``*-nproc.conf`` file contents."""
+    def __init__(cls, *args, **kwargs):
```
Comment (bfahr) [inline insights/parsers/nproc.py]: Same comment as above regarding use of `cls`.

Comment (xiangce) [reply insights/parsers/nproc.py]: @PaulWay - How do you think about the comments that Bob commented?

**Comment by PaulWay** [comment]
OK, I've fixed those issues - let me know what you think!

**Comment by xiangce** [comment]
@PaulWay - Since #819 is merged first, this PR should update correspondingly.

**Pull request merged**